projecttacoma/deqm-test-server-frontend
# Care gaps request

## Summary
Implementation of Care Gaps request preview, calculate button, and server request/response handling

## New behavior
- The "Calculate" button on the care-gaps page is conditionally enabled depending on whether the user has inputted the necessary inputs for a valid care gaps request.
- The request preview is now implemented - user's inputted values included in the preview as the user inputs them
- If the "Calculate" button is pressed, the request built in the preview is sent to the server for processing
- If the request is successful, a success notification is displayed for the user and the care gaps response is displayed in a Prism component
- If unsuccessful, an error notification is displayed

## Code changes
- **pages/[resource]/[id]/care-gaps.tsx** implements request preview and request sending 
- **__tests__/pages/resource/id/care-gaps.tsx** includes unit tests for new functionality
- **__tests__/pages/resource/id/evaluate.tsx** altered so unit tests use DateTime.now().year instead of hard-coded 2022

## Testing Guidance
- See the README.md for first time setup instructions.
#### Test in a browser
- To start the frontend: `npm run dev` then navigate to http://localhost:3001 in a browser
1. Valid navigation:
    - Click on “Measure” from the left-hand-side navigation menu. Click on any of the measure IDs. Click the “Calculate Care Gaps” button
    - The request preview and calculate button should be rendered (no "Placeholder" labels)
1. Successful care-gaps evaluation:
    - Choose a valid set of options to build the care-gaps request:
      a. Choose reportType: Subject, then select a Patient  
               OR
      b. Chose reportType: Organization, then select an Organization
    - Click the Calculate button. A success notification and a Prism component with the care-gaps report should both render
2. Failed care-gaps evaluation:
    - Create a new Measure resource with no body (from the Measure home, using the Create button).
      - Use: ` {"resourceType": "Measure"}`
    - On the care-gaps page for that new measure, click the Calculate button and an error notification should appear, specifying the issue.

#### Unit tests
- Run the unit tests: `npm run test`
